pkg/service/providerindex/providerindex.go

```diff
@@ -134,8 +192,58 @@ func (pi *ProviderIndex) filterBySpace(results []model.ProviderResult, mh mh.Mul
 // Publish should do the following:
 // 1. Write the entries to the cache with no expiration until publishing is complete
 // 2. Generate an advertisement for the advertised hashes and publish/announce it
-func (pi *ProviderIndex) Publish(context.Context, []mh.Multihash, model.ProviderResult) {
+func (pi *ProviderIndex) Publish(ctx context.Context, provider *peer.AddrInfo, contextID string, digests []mh.Multihash, meta meta.Metadata) error {
+	mdb, err := meta.MarshalBinary()
+	if err != nil {
+		return fmt.Errorf("marshaling metadata: %w", err)
+	}
+
+	pr := model.ProviderResult{
+		ContextID: []byte(contextID),
+		Metadata:  mdb,
+		Provider:  provider,
+	}
+
+	q := jobqueue.NewJobQueue(
```

Comment:
Yeah...I wanted to use it to understand how it works and also dogfood it within the project. Might be overkill...perhaps it's better to walk the chain to see how many entries we need to update and then decide on the concurrency? IDK, would be nice if the jobqueue spawned workers as needed instead of upfront.